lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -1289,15 +1289,17 @@ bool DiagnosticVerifier::finishProcessing() {
 
   SmallVector<unsigned, 4> additionalBufferIDs;
   for (auto path : AdditionalFilePaths) {
-    auto bufferID = SM.getIDForBufferIdentifier(path);
+    llvm::SmallString<261> buffer{path};
+    llvm::sys::path::make_preferred(buffer);
+    auto bufferID = SM.getIDForBufferIdentifier(buffer);
```

Comment:
Should there be assertions in central locations like `getIDForBufferIdentifier()` to catch code that uses non-preferred paths? Or maybe a new type or something? I honestly find it a little difficult to believe that there are only four places where we need to do this.

Reply:
I don't know what that central location is, this has been a long standing issue. This fixes the issues that are being uncovered here by enabling some of the tests that you added a long time ago.